### PR TITLE
Change warning options in streamly and streamly-core packages

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fixes:https://github.com/composewell/streamly/issues/1748

File: streamly.cabal [hunk 1]
```diff
@@ -669,7 +671,7 @@ library
                      , ghc-prim          >= 0.2   && < 0.10
                      , mtl               >= 2.2   && < 2.3
                      , transformers      >= 0.4   && < 0.7
-                     , filepath          >= 1.2.0.0 && < 1.5
+                     --, filepath          >= 1.2.0.0 && < 1.5
```
Comment (harendra-kumar) [inline streamly.cabal:674]: Just remove instead of commenting it out.

File: core/streamly-core.cabal [hunk 2]
```diff
@@ -353,12 +356,12 @@ library
                        base              >= 4.9   && < 4.18
                      , containers        >= 0.5   && < 0.7
                      , deepseq           >= 1.4.1 && < 1.5
-                     , directory         >= 1.2.2 && < 1.4
+                     --, directory         >= 1.2.2 && < 1.4
```
Comment (harendra-kumar) [inline core/streamly-core.cabal:359]: Just remove instead of commenting it out.

File: core/streamly-core.cabal [hunk 3]
```diff
@@ -353,12 +356,12 @@ library
...
                      , exceptions        >= 0.8   && < 0.11
                      , ghc-prim          >= 0.2   && < 0.10
                      , mtl               >= 2.2   && < 2.3
                      , transformers      >= 0.4   && < 0.7
-                     , filepath          >= 1.2.0.0 && < 1.5
+                     --, filepath          >= 1.2.0.0 && < 1.5
```
Comment (harendra-kumar) [inline core/streamly-core.cabal:364]: Just remove instead of commenting it out.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**